ambry-messageformat/src/main/java/com.github.ambry.messageformat/MessageFormatRecord.java

```diff
@@ -657,7 +658,88 @@ public static void serializeMetadataContentRecord(ByteBuffer outputBuffer, List<
         StoreKey storeKey = storeKeyFactory.getStoreKey(stream);
         keys.add(storeKey);
       }
-      return keys;
+      return new MultiPartMetadata(keys);
+    }
+  }
+
+  /**
+   *  - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+   * |         |              |              |            |            |          |
+   * | version |  chunk size  |  total size  |    key1    |     key2   |  ......  |
+   * |(2 bytes)|  (4 bytes)   |  (8 bytes)   |            |            |  ......  |
+   * |         |              |              |            |            |          |
+   *  - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+   *  version         - The version of the metadata content record
+   *
+   *  max chunk size  - maximum size of a data chunk in bytes
```

Comment:
sorry, I changed the field name to chunk size at some point and forgot to fix the comment.